### PR TITLE
bpo-36984: Improve version added references in `typing` module docs

### PR DESCRIPTION
**Pull request by asottile**
https://bugs.python.org/issue36984

### CODE REVIEW

File: Doc/library/typing.rst [hunk 1]
```diff
@@ -599,6 +599,7 @@ The module defines the following classes, functions and decorators:
 
    A generic version of :class:`collections.deque`.
 
+   .. versionadded:: 3.5.4
    .. versionadded:: 3.6.1
```
Comment (Carreau) [inline Doc/library/typing.rst:603]: Why is there 2 "versionadded" here ? was this added both ein 3.5.4 and 3.6.1 ? Meaning it's not in 3.6.0 ?
If that's the case that might look like a typo in the docs from a casual reader, and might need a  clarification paragraph.

File: Doc/library/typing.rst [hunk 2]
```diff
@@ -708,12 +719,14 @@ The module defines the following classes, functions and decorators:
 
    A generic version of :class:`collections.Counter`.
 
+   .. versionadded:: 3.5.4
    .. versionadded:: 3.6.1
```
Comment (Carreau) [inline Doc/library/typing.rst:723]: Same on all of these; that feel super weird;

**Review by Carreau — APPROVED** [review]
Version added are always super useful.

File: Doc/library/typing.rst (hunk 1 above)
Comment (asottile) [reply Doc/library/typing.rst:603]: that's correct! it _**is**_ pretty weird!

Comment (ilevkivskyi) [reply Doc/library/typing.rst:603]: Yeah, the story with versions is related to the fact that `typing` module used to be provisional, so features were added at random time, even between minor releases.

**Review by ilevkivskyi — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**